research/picai/preprocessing.py

```diff
@@ -0,0 +1,358 @@
+from __future__ import annotations
+
+import os
+from abc import ABC, abstractmethod
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass
+from functools import partial, reduce
+from pathlib import Path
+from typing import List, Optional, Sequence, Tuple
+
+import numpy as np
+import SimpleITK as sitk
+from preprocessing_transforms import crop_or_pad, resample_img
+
+
+@dataclass
+class PreprocessingSettings:
+    def __init__(
+        self,
+        scans_write_dir: Path,
+        annotation_write_dir: Path,
+        size: Optional[Sequence[int]],
```

Comment:
Are size and spacing of a fixed length? If so, maybe we make these `Tuple[int, int, int]` or whatever to enforce that?